lib/src/modules/task.cc

```diff
@@ -453,9 +453,9 @@ ActionResponse Task::callAction(const ActionRequest& request)
 {
     auto task_execution_params = request.params();
     auto task_input_method = task_execution_params.includes("input_method") ?
-        task_execution_params.get<std::string>("input_method") : std::string{""};
+        task_execution_params.get<std::string>("input_method") : std::string{"both"};
```

Comment:
Applying this default _here_ breaks the line where we set `powershell` as the default for `.ps1` files 

https://github.com/puppetlabs/pxp-agent/blob/1.9.x/lib/src/modules/task.cc#L473

I think we probably want to keep it as an empty string _here_ and then set `both` just _after_ we would potentially set `powershell`. So that code would become: if it has no default, then set it to `powershell` if it's `.ps1` and `both` otherwise.

That would also address the _other_ issue this introduces which is that if the task metadata explicitly sets `"input_method": ""`, it was previously treated as `both` but will now be treated as an unknown input method. Admittedly, there's no good _reason_ to set it to that, but it is a behavior change. I will check what Bolt does in that case. An error may actually be consistent.

Reply:
Ah, that is what i get for updating this with half my attention on it. :flushed: will update when i can actually focus on it.